index.bs

```diff
@@ -269,88 +304,146 @@ which runs these steps:
+when a blob is backed by a file on disk (i.e. was created by the
+[=create a file backed File object=] algorithm).
+User agents can use modification time stamps and other mechanisms to maintain
+this requirement, but this is left as an implementation detail.
+
+### Byte Sequence backed blobs ### {#byte-sequence-backed-blobs}
+
+The [=snapshot state=] for a byte sequence backed blob contains a
+<dfn for="bytes blob snapshot state"><code>"data"</code></dfn> member, a [=byte sequence=].
+
+<div algorithm="create bytes blob">
+To <dfn lt="creating blob data from bytes">create blob data from bytes</dfn>
+given |bytes| (a [=byte sequence=]), run the following steps:
+
+1. Let |blob data| be a new [=blob data description=].
+1. Set |blob data|.[=snapshot state=][<code><a for="bytes blob snapshot state">"data"</a></code>] to |bytes|.
+
+1. Set |blob data|.[=read initialization algorithm=] to the [=bytes blob read initialization steps=].
+1. Set |blob data|.[=read algorithm=] to the [=bytes blob read steps=].
 
-1. Return a {{Blob}} object referring to |bytes| as its associated <a>byte</a> sequence,
-  with its {{Blob/size}} set to the length of |bytes|,
-  and its {{Blob/type}} set to the value of |t| from the substeps above.
+1. Return |blob data|.
 
 </div>
 
-### Constructor Parameters ### {#constructorParams}
+A <dfn>bytes blob read state</dfn> is a [=struct=] conssting of
+<dfn for="bytes blob read state">bytes</dfn> (a [=byte sequence=]).
 
-The {{Blob()}} constructor can be invoked with the parameters below:
+<div algorithm>
+The <dfn>bytes blob read initialization steps</dfn>,
+given a |snapshot state| and |offset| are:
 
-<dl>
-  <dt>A <dfn id="dfn-blobParts" argument for="Blob/Blob(blobParts, options)"><code>blobParts</code></dfn> <code>sequence</code>
-  <dd>which takes any number of the following types of elements, and in any order:
-    * {{BufferSource}} elements.
-    * {{Blob}} elements.
-    * {{USVString}} elements.
-
-  <dt id="dfn-BlobPropertyBagMembers">An *optional* {{BlobPropertyBag}}
-  <dd>which takes these optional members:
-    * <dfn id="dfn-BPtype" dict-member for="BlobPropertyBag">type</dfn>,
-      the ASCII-encoded string in lower case representing the media type of the {{Blob}}.
-      Normative conditions for this member are provided in the [[#constructorBlob]].
-    * <dfn dict-member for="BlobPropertyBag">endings</dfn>,
-      an enum which can take the values {{"transparent"}} or {{"native"}}.
-      By default this is set to {{"transparent"}}. If set to {{"native"}},
-      [=convert line endings to native|line endings will be converted to native=]
-      in any {{USVString}} elements in {{blobParts}}.
-</dl>
+1. Let |read state| be a new [=bytes blob read state=].
+1. If |offset| is larger than |snapshot state|[<code><a for="bytes blob snapshot state">"data"</a></code>]'s [=byte sequence/length=],
+  set |read state|.[=bytes blob read state/bytes=] to an empty [=byte sequence=].
```

Comment:
FWIW, I don't see the _struct_._name_ syntax called out in Infra. Do you know if there's plan to add it?